noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/partial_note.nr

```diff
@@ -93,35 +89,26 @@ pub fn compute_partial_public_log_payload<let N: u32, let M: u32>(
     // Note: the aes128_encrypt builtin fn automatically appends bytes to the
     // input, according to pkcs#7; hence why the output `header_ciphertext_bytes` is 16
     // bytes larger than the input in this case.
-    let header_ciphertext_bytes = aes128_encrypt(header_plaintext, iv, sym_key);
-    // I recall that converting a slice to an array incurs constraints, so I'll check the length this way instead:
-    assert(header_ciphertext_bytes.len() == HEADER_CIPHERTEXT_SIZE_IN_BYTES);
+    let header_ciphertext_bytes: [u8; HEADER_CIPHERTEXT_SIZE_IN_BYTES] =
+        aes128_encrypt(header_plaintext, iv, sym_key);
 
     // *****************************************************************************
     // Prepend / append more bytes of data to the ciphertext, before converting back
     // to fields.
     // *****************************************************************************
 
-    let mut log_bytes_padding_to_mult_31 = get_arr_of_size__log_bytes_padding__from_PT::<2 + N>();
+    let mut log_bytes_padding_to_mult_31 =
+        get_arr_of_size__log_bytes_padding__from_plaintext(final_plaintext);
+    /// Safety: this randomness won't be constrained to be random. It's in the
+    /// interest of the executor of this fn to encrypt with random bytes.
     log_bytes_padding_to_mult_31 = unsafe { get_random_bytes() };
 
-    let mut log_bytes = get_arr_of_size__log_bytes__from_PT::<2 + N>();
```

Comment:
@nventuro there was a bug in the code I merged the other day. This should be `N`; not `2 + N`. The `2 + N` is from the previous log encoding. I guess we got lucky (or unlucky) with the size of `N` in all of our tests not triggering a case where `2 + N` resulted in a different amount of padding. 
Anyway, this is (accidentally, but luckily) fixed in this PR. People might run into issues of partial notes not syncing until this fix is merged.